TestProject.BL.Test/Services/UserServiceTests.cs

```diff
@@ -75,7 +75,34 @@ public void Should_get_user_profile(string email, UserProfile expected)
             _mockRatingRepository.Setup(repo => repo.Get(It.IsAny<Func<PostRating, bool>>())).Returns(new List<PostRating> { rating });
             _mockUserProfileMapper.Setup(mapper => mapper.ToBlModel(user)).Returns(new UserProfile { Email = email });
 
-            var result = _userService.GetProfile(email);
+            var result = _userService.GetByEmail(email);
+
+            result.Should().BeEquivalentTo(expected);
+        }
+
+        [Fact]
+        public void If_current_user_is_not_found_should_throw_exception()
+        {
+            var users = new List<User>();
+            _mockUserRepository.Setup(repo => repo.Get(It.IsAny<Func<User, bool>>()))
+                .Returns((Func<User, bool> predicate) => users.Where(predicate).ToList());
+
+            Assert.Throws<UserNotFoundException>(() => _userService.GetByEmail("email"));
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUser))]
+        public void Should_get__user_profile(int userId, UserProfile expected)
+        {
+            var user = new User { Id = userId };
+            var users = new List<User>() { user };
+            var rating = new PostRating() { Value = RatingValue.Plus };
+            _mockUserRepository.Setup(repo => repo.Get(It.IsAny<Func<User, bool>>()))
+                .Returns((Func<User, bool> predicate) => users.Where(predicate).ToList());
+            _mockRatingRepository.Setup(repo => repo.Get(It.IsAny<Func<PostRating, bool>>())).Returns(new List<PostRating> { rating });
```

Comment:
It will always return the same value. Make it dependent on implementation.
:memo: